fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -1074,8 +1074,15 @@ private enum BinaryPhysicalOperator {
 
         @Nullable
         public Object eval(@Nullable final Object arg1, @Nullable final Object arg2) {
+            // handle nulls
             if (arg1 == null || arg2 == null) {
-                return null;
+                if (type == Comparisons.Type.IS_DISTINCT_FROM) {
+                    return arg1 != null || arg2 != null;
+                } else if (type == Comparisons.Type.NOT_DISTINCT_FROM) {
+                    return arg1 == null && arg2 == null;
+                } else {
+                    return null;
+                }
```

Comment:
It is interesting that we have this early treatment of `null` values here instead of delegating it to the actual underlying physical operator.

I opened an issue to fix this, as well as introducing `NULL` type code handling (instead of the deprecated `UNKNOWN`).
https://github.com/FoundationDB/fdb-record-layer/issues/3526